src/components/EditInput/index.jsx

```diff
@@ -0,0 +1,139 @@
+import { Link } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { updateMyInfo } from "../../apis/api";
+
+export const MyPageInput = ({ content }) => {
+  const [isEmailEdit, setIsEmailEdit] = useState(false);
+  const [isUsernameEdit, setIsUsernameEdit] = useState(false);
+  const [isCollegeEdit, setIsCollegeEdit] = useState(false);
+  const [isMajorEdit, setIsMajorEdit] = useState(false);
+  const [formData, setFormData] = useState(content);
+
+  const handleChange = (e) => {
+
+    if (e.target.id === "email" || e.target.id === "username")
+      {
+        setFormData({ ...formData, [`user`]: {...formData.user, [e.target.id]: e.target.value} });
+      }
+    else setFormData({ ...formData, [e.target.id]: e.target.value });
+  };
+
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    await updateMyInfo(formData);
+    setIsEmailEdit(false);
+    setIsUsernameEdit(false);
+    setIsCollegeEdit(false);
+    setIsMajorEdit(false);
+  };
+
+  const onEmailEditClick = async (e) => {
+    setIsEmailEdit(true);
+  };
+
+  const onUsernameEditClick = async (e) => {
+    setIsUsernameEdit(true);
+  };
+
+  const onCollegeEditClick = async (e) => {
+    setIsCollegeEdit(true);
+  };
+
+  const onMajorEditClick = async (e) => {
+    setIsMajorEdit(true);
+  };
+
```

Comment:
우선, 이 부분은 그냥 setState를 활용해서 state를 변경해주는 것이기 때문에 비동기 처리가 아니예요! async를 사용할 필요가 없습니다.
그리고 e를 사용하지 않는다면 굳이 적어둘 필요 없습니다!

또, 이 부분도 `isEdit`으로 한번에 관리한다면 훨씬 편하겠죠?
만약에 button의 id를 email, username 이런식으로 설정한다고 가정한다면

```suggestion
  const onEditClick = (e) => {
   const { id } = e.target
   setIsEdit((prev) => ({...prev, [id] : true })
  };
```

이런식으로요!